File: app/src/main/java/com/dumuzeyn/mp3player/FullPlayerPageOrder.java
package com.dumuzeyn.mp3player;

final class FullPlayerPageOrder {
    static final int PLAYER = 0;
    static final int LYRICS = 1;
    static final int QUEUE = 2;

    private FullPlayerPageOrder() {
    }

    static int afterRightSwipe(int position) {
        return Math.max(PLAYER, position - 1);
    }

    static int afterLeftSwipe(int position) {
        return Math.min(QUEUE, position + 1);
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/FullPlayerPageOrder.kt
package com.dumuzeyn.mp3player

object FullPlayerPageOrder {
    const val PLAYER = 0
    const val LYRICS = 1
    const val QUEUE = 2

    @JvmStatic
    fun afterRightSwipe(position: Int): Int = position.coerceAtLeast(PLAYER) -
        if (position > PLAYER) 1 else 0

    @JvmStatic
    fun afterLeftSwipe(position: Int): Int = (position + 1).coerceAtMost(QUEUE)
}
